src/promptflow/promptflow/_utils/tool_utils.py

```diff
@@ -346,6 +364,16 @@ def _get_function_path(function):
     return func, func_path
 
 
+def _find_template_variable_index(template_str, variable):
+    matches = re.finditer(r'\{([^}]*)\}', template_str)
```

Comment:
Can we move this line out of this function, and pass the matches into the function. Otherwise, this finditer will be execute for every variable.

Reply:
moved, thanks.